[3/N] [DTensor device order] Make some placement type class method static

zpcore opened the pull request:
Some methods in `Placement` class can be exposed as static.

Those method should be useful w/o initializing the object. E.g., when we `distribute_tensor` from normal tensor, we may want:
```
local_tensor = Shard.shard_tensor(tensor_dim, local_tensor, device_mesh, mesh_dim,)
```

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #165197
* #165092
* #164902
* #164821
* __->__ #164820
* #164806
* #164797



cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @pragupta @msaroufim @dcci

zpcore committed:
[3/N] [DTensor device order] Make some placement type class method static

[ghstack-poisoned]

zpcore committed:
Update on "[3/N] [DTensor device order] Make some placement type class method static"

Some methods in `Placement` class can be exposed as static.




cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

zpcore committed:
Update on "[3/N] [DTensor device order] Make some placement type class method static"

Some methods in `Placement` class can be exposed as static.




cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

zpcore committed:
Update on "[3/N] [DTensor device order] Make some placement type class method static"

Some methods in `Placement` class can be exposed as static.




cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

zpcore committed:
Update on "[3/N] [DTensor device order] Make some placement type class method static"

Some methods in `Placement` class can be exposed as static.




cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

fduwjj reviewed (approved):
No need to do it in this PR, but if we can add some comments to this file that will be really good. IIUC, most of collective calls happen in this `placement_types.py` so when users start to investigate DTensor collective issues, for example, integration with torchcomm, people know where to start.

zpcore committed:
Update on "[3/N] [DTensor device order] Make some placement type class method static"

Some methods in `Placement` class can be exposed as static.




cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

zpcore committed:
Update on "[3/N] [DTensor device order] Make some placement type class method static"

Some methods in `Placement` class can be exposed as static.




cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

zpcore committed:
Update on "[3/N] [DTensor device order] Make some placement type class method static"

Some methods in `Placement` class can be exposed as static.




cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

wanchaol reviewed (changes requested):
Can you give some motivations on this change (in the PR summary)?

The consequence of this change is that it exposes these methods as public methods in the Placement, so we would need to maintain the BC of these APIs going forward and could not change them easily under the hood. I would not prefer to expose them unless there's a strong motivation

zpcore committed:
Update on "[3/N] [DTensor device order] Make some placement type class method static"

Some methods in `Placement` class can be exposed as static.

Those method should be useful w/o initializing the object. E.g., when we `distribute_tensor` from normal tensor, we may want:
```
local_tensor = Shard.shard_tensor(tensor_dim, local_tensor, device_mesh, mesh_dim,)
```




cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

zpcore committed:
Update on "[3/N] [DTensor device order] Make some placement type class method static"

Some methods in `Placement` class can be exposed as static.

Those method should be useful w/o initializing the object. E.g., when we `distribute_tensor` from normal tensor, we may want:
```
local_tensor = Shard.shard_tensor(tensor_dim, local_tensor, device_mesh, mesh_dim,)
```




cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

zpcore committed:
Update on "[3/N] [DTensor device order] Make some placement type class method static"

Some methods in `Placement` class can be exposed as static.

Those method should be useful w/o initializing the object. E.g., when we `distribute_tensor` from normal tensor, we may want:
```
local_tensor = Shard.shard_tensor(tensor_dim, local_tensor, device_mesh, mesh_dim,)
```




cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

zpcore committed:
Update on "[3/N] [DTensor device order] Make some placement type class method static"

Some methods in `Placement` class can be exposed as static.

Those method should be useful w/o initializing the object. E.g., when we `distribute_tensor` from normal tensor, we may want:
```
local_tensor = Shard.shard_tensor(tensor_dim, local_tensor, device_mesh, mesh_dim,)
```




cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

zpcore committed:
Update on "[3/N] [DTensor device order] Make some placement type class method static"

Some methods in `Placement` class can be exposed as static.

Those method should be useful w/o initializing the object. E.g., when we `distribute_tensor` from normal tensor, we may want:
```
local_tensor = Shard.shard_tensor(tensor_dim, local_tensor, device_mesh, mesh_dim,)
```




cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]
